develop/reference/articles/06-themes/02-theme-components-and-workflow.markdown

```diff
@@ -0,0 +1,238 @@
+  theme:
+
+      <#if has_navigation && is_setup_complete>
+        <button aria-controls="navigationCollapse" aria-expanded="false" 
+        aria-label="Toggle navigation" class="navbar-toggler navbar-toggler-right" 
+        data-target="#navigationCollapse" data-toggle="collapse" type="button">
+          <span class="navbar-toggler-icon"></span>
+        </button>
+
+        <div aria-expanded="false" class="collapse navbar-collapse" id="navigationCollapse">
+          <@liferay.navigation_menu default_preferences="${preferences}" />
+        </div>
+      </#if>
+
+![Figure 1: The collapsed navbar provides simplified user-friendly navigation for mobile devices.](../../images/portal-layout-mobile-nav.png)
+
+- `portal_pop_up.ftl`: The theme template for pop up dialogs for the theme's 
+  portlets. Similar to `portal_normal.ftl`, `portal_pop_up.ftl` provides the 
+  markup template for all pop-up dialogs, such as a portlet's Configuration menu 
+  (includes `init.ftl`, which includes `init_custom.ftl`).
+
+![Figure 2: Each theme template provides a portion of the page's markup and functionality.](../../images/portal-layout-theme-templates.png)
+
+### Theme Template Utilities [](id=theme-template-utilities)
+
+Liferay provides several FreeMarker variables and macros that you can use in 
+your theme templates to include portlets, use taglibs, access theme objects, and 
+more. You can see examples of these used in `portal_normal`. These utilities are 
+included in the files listed below:
+
+- [`Init.ftl`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/frontend-theme/frontend-theme-unstyled/src/main/resources/META-INF/resources/_unstyled/templates/init.ftl): 
+  Provides access to common theme variables
+- [`FTL_Liferay.ftl`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl): 
+  Provides macros for commonly used portlets and theme resources. See the 
+  [Macros tutorial](/develop/tutorials/-/knowledge_base/7-1/using-liferays-macros-in-your-theme) 
+  for more information.
+- [`taglib-mappings.properties`](https://github.com/liferay/liferay-portal/blob/7.1.x/modules/apps/frontend-taglib/frontend-taglib-util-freemarker-contributor/src/main/resources/META-INF/taglib-mappings.properties): 
+  Maps the portal taglibs to names for FreeMarker macros. Taglibs let you 
+  quickly create common UI components. This properties file is also provided 
+  separately for each app taglib. See the 
+  [Taglib tutorials](/develop/tutorials/-/knowledge_base/7-1/front-end-taglibs) 
+  for more information on using each taglib in your theme templates.
+
+## CSS Frameworks and Extensions [](id=css-frameworks-and-extensions)
+
+As noted above, @product@ supports the SASS CSS extension. CSS files that use 
+SASS should be prefixed with `_`, such as `_custom.scss` (where custom styles 
```

Comment:
Not sure if we need this in the docs? The only reason we prefix with underscore is so that Sass knows not to output a CSS duplicate of the file (e.g., if there was a `custom.scss`, Sass would output a `custom.css`). Sass also won't do anything with a partial unless you `@import` it.

Sometimes you want a Sass file without an underscore so you can do: `<link rel="stylesheet" href="custom.css" />` at the top of your page. https://sass-lang.com/guide provides a better description of Sass partials under the Partials section.

Reply:
Thanks @pat270 ! I'll clarify that in the docs, or leave it out and just link to sass-lang guide you provided.

Reply:
I opted to leave that information out, as you suggested. Thanks!